src/track.c

```diff
@@ -261,15 +294,41 @@ void tracking_channel_update(u8 channel)
       chan->code_phase_rate_fp = chan->code_phase_rate
         * NAP_TRACK_CODE_PHASE_RATE_UNITS_PER_HZ;
 
-      chan->carrier_freq_fp_prev = chan->carrier_freq_fp;
       chan->carrier_freq_fp = chan->carrier_freq
         * NAP_TRACK_CARRIER_FREQ_UNITS_PER_HZ;
 
+      if (chan->int_ms != 1) {
+        s32 I = (cs[1].I - chan->alias_detect.first_I) / (LONG_INTEGRATION_INTERVAL-1);
+        s32 Q = (cs[1].Q - chan->alias_detect.first_Q) / (LONG_INTEGRATION_INTERVAL-1);
```

Comment:
I don't really think so.  This is a consequence of the fact that cs[] are the sum of the short and long interval accumulated in tracking_channel_get_corrs(). We could also keep extra state of the intermediate values around.  This is working, so I'd rather just leave it alone.  Either way, I think the alias detect functions in libswiftnav are fine.